### DISCUSSION TITLE
Thoughts on adding Typescript

### DISCUSSION THREAD

**Discussion by jshthornton** [bhubr/react-simple-oauth2-login#48] [Ideas]
Adding Typescript would help reduce bugs, not require unit tests that just check for input vs. output, as well as produce a definition file upon build that will allow consuming apps to know props.
This would also cut down on bundle size as the `prop-types` dependency can be dropped.

**Comment by bhubr** [comment]
Funny that you should mention this now: I've been considering this since a few days, since I'm getting more in more into TS. Consider me sold on the idea!
